src/mcpdotnet/Utils/Json/JsonSerializerOptionsExtensions.cs

```diff
@@ -1,32 +1,67 @@
-﻿using System.Text.Json;
+﻿using McpDotNet.Protocol.Messages;
+using McpDotNet.Protocol.Types;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Nodes;
 using System.Text.Json.Serialization;
+using System.Text.Json.Serialization.Metadata;
 
 namespace McpDotNet.Utils.Json;
 
 /// <summary>
 /// Extensions for configuring System.Text.Json serialization options for MCP.
 /// </summary>
-internal static class JsonSerializerOptionsExtensions
+internal static partial class JsonSerializerOptionsExtensions
 {
     public static JsonSerializerOptions DefaultOptions { get; } = CreateDefaultOptions();
 
     /// <summary>
     /// Creates default options to use for MCP-related serialization.
     /// </summary>
     /// <returns>The configured options.</returns>
+    [UnconditionalSuppressMessage("AotAnalysis", "IL3050", Justification = "DefaultJsonTypeInfoResolver is only used when reflection-based serialization is enabled")]
+    [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", Justification = "DefaultJsonTypeInfoResolver is only used when reflection-based serialization is enabled")]
     private static JsonSerializerOptions CreateDefaultOptions()
     {
-        JsonSerializerOptions options = new();
+        // If reflection-based serialization is enabled by default, use it, as it's the most permissive in terms of what it can serialize,
+        // and we want to be flexible in terms of what can be put into the various collections in the object model.
+        // Otherwise, use the source-generated options to enable trimming and Native AOT.
 
-        // Add custom converters
-        options.Converters.Add(new JsonRpcMessageConverter());
-        options.Converters.Add(new JsonStringEnumConverter(JsonNamingPolicy.CamelCase));
+        if (JsonSerializer.IsReflectionEnabledByDefault)
```

Comment:
Have you tried running the tests with this disabled?

Reply:
Not really, and we haven't really done this in MEAI either. In runtime it's easy using the `RemoteExecutor` class but here we probably need to wire AOT testing. I figured we should just turn on the analyzer for now to get feedback on the currently available APIs.

Reply:
I meant just commenting out this if block. Wouldn't we expect all tests to still pass, with no other infra changes, and if they don't, that means we're missing something in the source generated context?

Reply:
Commenting out as in deleting the context and have it throw in that mode?

Reply:
Commented out as in delete the if block locally, run the tests, verify they all pass, put back the if block. Just a quick local sanity check.